wagtailcache/cache.py

```diff
@@ -412,10 +408,9 @@ def _wrapped_view_func(
     ) -> HttpResponse:
         # Try to fetch an already cached page from wagtail-cache.
         response = FetchFromCacheMiddleware().process_request(request)
-        if response:
-            return response
-        # Since we don't have a response at this point, process the request.
-        response = view_func(request, *args, **kwargs)
+        if response is None:
```

Comment:
Looks like this change will reduce performance of the cache reads slightly. Because now, it is fetching from cache (assume the cache is filesystem or network... therefore slow), then it is going to save that response back into the cache (line 415). So this will effectively double the time to process a cache read.

Ideally cache reads should have performance as top priority.